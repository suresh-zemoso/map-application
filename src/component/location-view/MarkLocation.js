import React from 'react';
import BaseMapView from '../public-map/BaseMapView';
import mapboxgl from 'mapbox-gl';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';


export default function MarkLocation(props) {

    const { id } = useParams();
    const locations = useSelector(state => state.locations);
    const mapAttribute = locations.find(location => {
        return location.id == id
    })
    function flyToLocation(map) {
        map.on('load', (e) => {
            map.flyTo({
                center: [
                    mapAttribute.lng,
                    mapAttribute.lat,
                ],
                zoom: 7,
                essential: true // this animation is considered essential with respect to prefers-reduced-motion
            });

            var marker = new mapboxgl.Marker();
            // Update the data to a new position based on the animation timestamp. The
            // divisor in the expression `timestamp / 1000` controls the animation speed.
            marker.setLngLat([
                mapAttribute.lng,
                mapAttribute.lat
            ]);

            // Ensure it's added to the map. This is safe to call if it's already added.
            marker.addTo(map);

            // Request the next frame of the animation.
            // requestAnimationFrame(animateMarker);
        });
    }



    return (
        <BaseMapView mapAttribute={{ lng: 5, lat: 34, zoom: 2 }} handlers={[flyToLocation]} />
    )
}
