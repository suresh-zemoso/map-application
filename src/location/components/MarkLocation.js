import React from 'react';
import BaseMapView from '../../location/components/BaseMapView';
import mapboxgl from 'mapbox-gl';
import { useParams } from 'react-router-dom';
import { useSelector } from 'react-redux';



const defaultProps = {

}

const propTypes = {


}
const MarkLocation = (props) => {

    const { id } = useParams();
    const locations = useSelector(state => state.locationState.locations);
    const mapAttribute = locations.find(location => {
        return location.id == id
    })


    function onLoadListener(map, e) {
        map.flyTo({
            center: [
                mapAttribute.lng,
                mapAttribute.lat,
            ],
            zoom: 7,
            essential: true // this animation is considered essential with respect to prefers-reduced-motion
        });

        var marker = new mapboxgl.Marker();
        marker.setLngLat([
            mapAttribute.lng,
            mapAttribute.lat
        ]);
        // Ensure it's added to the map. This is safe to call if it's already added.
        marker.addTo(map);
    }

    const handlers =
        [{
            addEventListener: (map) => map.on('load', (e) => onLoadListener(map, e)),
            removeEventListener: (map) => map.off('load', (e) => onLoadListener(map, e))
        }]


    return (
        <BaseMapView mapAttribute={{ lng: 5, lat: 34, zoom: 2 }} handlers={handlers} />
    )
}

MarkLocation.propTypes = propTypes;
MarkLocation.defaultProps = defaultProps;

export default MarkLocation;