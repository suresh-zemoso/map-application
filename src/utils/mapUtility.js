function getGeoCoder(mapboxgl) {
    return new window.MapboxGeocoder({
        accessToken: mapboxgl.accessToken,
        mapboxgl: mapboxgl
    })
}

function getGeoLocateControl(mapboxgl) {
    return new mapboxgl.GeolocateControl({
        positionOptions: {
            enableHighAccuracy: true
        },
        trackUserLocation: true
    })
}

export const mapUtility = { getGeoCoder, getGeoLocateControl }