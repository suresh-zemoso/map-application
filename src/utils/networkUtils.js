import axios from 'axios';

export function handleApiResponse(response) {


    const data = response && response.data;

    if (response.status !== 200 && response.status !== 201) {
        if (response.status === 401) {
            console.debug("Unauthorized got excecuted")
        }
        const error = (data && data.message) || response;
        return Promise.reject(error);
    }

    return data;

}

axios.interceptors.response.use((response) => {
    return response;
}, function (error) {
    // Do something with response error
    return Promise.reject(error.response);
});