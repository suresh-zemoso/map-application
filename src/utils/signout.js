import { logout } from "../auth/actions/loginAction"
import { store } from "../store";

export default function signout(props) {
    store.dispatch(logout())
}
