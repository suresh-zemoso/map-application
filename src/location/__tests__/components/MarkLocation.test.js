import React from 'react';
import { mount, shallow } from 'enzyme';
import MarkLocation from '../../components/MarkLocation';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mapUtility } from '../../../utils/mapUtility'

jest.mock('mapbox-gl', () => {
    const mockMap = jest.fn(() => ({
        on: jest.fn(),
        off: jest.fn(),
        getSource: jest.fn(),
        getLayer: jest.fn(),
        addControl: jest.fn(),
        addSource: jest.fn(),
        addLayer: jest.fn(),
        addImage: jest.fn(),
        loadImage: jest.fn(),
        hasImage: jest.fn(),
        dragPan: {
            enable: jest.fn(),
            disable: jest.fn()
        },
        scrollZoom: {
            enable: jest.fn(),
            disable: jest.fn()
        },
        doubleClickZoom: {
            enable: jest.fn(),
            disable: jest.fn()
        },
        touchZoomRotate: {
            enable: jest.fn(),
            disable: jest.fn()
        },
        setStyle: jest.fn(),
        queryRenderedFeatures: jest.fn(),
        getCenter: jest.fn(),
        flyTo: jest.fn(),
        isStyleLoaded: jest.fn(),
        getBounds: jest.fn(),
        getZoom: jest.fn(),
        resize: jest.fn(),
        getStyle: jest.fn(() => ({
            layers: []
        })),
        isMoving: jest.fn(),
        isZooming: jest.fn(),
        isRotating: jest.fn(),
        removeControl: jest.fn(),
        remove: jest.fn()
    }));

    const mockPopup = jest.fn(() => ({
        setLngLat: jest.fn(() => mockPopup()),
        setDOMContent: jest.fn(() => mockPopup()),
        remove: jest.fn(() => mockPopup()),
        addTo: jest.fn(() => mockPopup()),
        setMaxWidth: jest.fn(() => mockPopup())
    }));

    const mockMarker = jest.fn(() => ({
        setLngLat: jest.fn(() => mockMarker()),
        remove: jest.fn(() => mockMarker()),
        addTo: jest.fn(() => mockMarker()),
        getElement: jest.fn(() => mockMarker()),
        on: jest.fn(),
        getLngLat: jest.fn()
    }));

    return {
        Map: mockMap,
        NavigationControl: jest.fn(),
        GeolocateControl: jest.fn(() => ({
            on: jest.fn(),
            trigger: jest.fn()
        })),
        Popup: mockPopup,
        LngLat: jest.fn((lat, lng) => ({
            lat, lng,
            wrap: jest.fn(() => ({
                lat, lng
            }))
        })),
        Marker: mockMarker,
        ScaleControl: jest.fn()
    };
});


describe('MarkLocation', () => {

    let useEffect;
    let store;
    let wrapper;

    const mockUseEffect = () => {
        useEffect.mockImplementationOnce(f => f());
    };

    beforeEach(() => {

        store = configureStore([thunk])({
            locationState: { locations: [] }
        });

        /* mocking useEffect */
        useEffect = jest.spyOn(React, "useEffect");
        mockUseEffect();

        jest.mock('react-router-dom', () => ({
            useParams: () => ({
                id: 1,
            }),
        }));
        jest.spyOn(mapUtility, "getGeoCoder").mockImplementation(() => { })

        /* shallow rendering */
        wrapper = mount(<Provider store={store}><MemoryRouter><MarkLocation /></MemoryRouter></Provider>);
    });

    const props = {
        mapMode: "satellite-v9",
        mapAttribute: { lng: 5, lat: 34, zoom: 2 },
        handlers: [],
        buttons: []
    };

    it('should render the MarkLocation Component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

});      