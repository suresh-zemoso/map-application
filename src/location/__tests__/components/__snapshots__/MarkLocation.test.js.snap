// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkLocation should render the MarkLocation Component correctly 1`] = `ReactWrapper {}`;
